File: CV/p5/code/multiclassLRTrain.py
import numpy as np

def multiclassLRTrain(x, y, param):

    classLabels = np.unique(y)
    numClass = classLabels.shape[0]
    numFeats = x.shape[0]
    numData = x.shape[1]

    # Initialize weights randomly (Implement gradient descent)
    model = {}
    model['w'] = np.random.randn(numClass, numFeats)*0.01
    model['classLabels'] = classLabels

    return model

File: CV/p5/code/multiclassLRTrain.m
function model = multiclassLRTrain(x, y, param)
classLabels = unique(y);
numClass = length(classLabels);
numFeats = size(x,1);
numData = size(x,2);

% Initialize weights randomly (Implement the gradient descent)
model.w = randn(numClass, numFeats)*0.01;
model.classLabels = classLabels;
